src/contexts/CartContext/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { ReactNode, createContext, useEffect, useState } from 'react'
+
+import { CartContextType, CartItem } from './CartContext.types'
+
+export const CartContext = createContext<CartContextType>({
+  cart: [],
+  isOpen: false,
+  totalPrice: 0,
+  totalItems: 0,
+  toggleCart: () => undefined,
+  addItem: () => undefined,
+  updateItem: () => undefined,
+  deleteItem: () => undefined,
+  clearCart: () => undefined,
+})
```

Comment:
Read my article once again and get rid of this terrifying empty state XD

it should be createContext<CartContextType | null>(null) at this point